lib/widgets/compose_box.dart

```diff
@@ -557,23 +557,22 @@ class _AttachFromCameraButton extends _AttachUploadsButton {
   }
 }
 
-/// The send button for _StreamComposeBox.
-class _StreamSendButton extends StatefulWidget {
-  const _StreamSendButton({
-    required this.narrow,
+class _SendButton extends StatefulWidget {
+  const _SendButton({
     required this.topicController,
     required this.contentController,
+    required this.getDestination,
   });
 
-  final StreamNarrow narrow;
-  final ComposeTopicController topicController;
+  final ComposeTopicController? topicController;
   final ComposeContentController contentController;
+  final MessageDestination Function() getDestination;
```

Comment:
> I could imagine it being convenient (in later work) to keep an up-to-date `MessageDestination` to be read in a build method

Sure, nothing wrong with that in general.

I think probably `SendableNarrow` will be more convenient — particularly because `DmNarrow` comes with handy other facilities on it, enabled by the invariants it maintains. That's why `_FixedDestinationContentInput` takes a `SendableNarrow`, rather than just a `MessageDestination`.

The reason the [MessageDestination] doc talks about [sendMessage], though, is just that [MessageDestination] is part of the API bindings, and [sendMessage] is the part of the API that it's for. It's similar to how the doc on [Anchor] says it's for [getMessages].

In particular the "in [sendMessage]" contrasts with how we specify a conversation when we talk to [getMessages], which is with an [ApiNarrow]. (Which needn't be a conversation, but very often is.)